scripts/triggers-end-to-end-tests/run.spec.js

```diff
@@ -335,3 +339,61 @@ describe("database and firestore emulator function triggers", function() {
     done();
   });
 });
+
+describe("pubsub emulator function triggers", function() {
+  var test;
+
+  before(function(done) {
+    this.timeout(TEST_SETUP_TIMEOUT);
+
+    expect(FIREBASE_PROJECT).to.not.be.an("undefined");
+    expect(FIREBASE_PROJECT).to.not.be.null;
+
+    async.series(
+      [
+        function(done) {
+          readConfig(function(err, config) {
+            if (err) {
+              done(new Error("error reading firebase.json: " + err));
+              return;
+            }
+            test = new TriggerEndToEndTest(config);
+            done();
+          });
+        },
+        function(done) {
+          test.startEmulators(["--only", "functions,pubsub"]);
+          /*
+           * Give some time for the emulator subprocesses to start up.
+           */
+          setTimeout(done, EMULATORS_STARTUP_DELAY_MS);
+        },
+      ],
+      done
+    );
+  });
+
+  it("should write to the pubsub emulator", function(done) {
+    this.timeout(EMULATOR_TEST_TIMEOUT);
+
+    test.writeToPubsub(function(err, response) {
+      expect(err).to.be.null;
+      expect(response.statusCode).to.equal(200);
+      setTimeout(done, EMULATORS_WRITE_DELAY_MS);
+    });
+  });
+
+  it("should have have triggered cloud functions", function(done) {
+    expect(test.pubsub_trigger_count).to.equal(1);
+    done();
+  });
+
+  after(function(done) {
+    this.timeout(EMULATORS_SHUTDOWN_DELAY_MS);
+    if (test) {
+      test.stopEmulators(done);
+      return;
+    }
+    done();
+  });
```

Comment:
Typically this goes after the `before` block, before the tests (`it`s). It's considered setup